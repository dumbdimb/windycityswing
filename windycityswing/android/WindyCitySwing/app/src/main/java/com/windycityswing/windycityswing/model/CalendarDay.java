package com.windycityswing.windycityswing.model;

import java.util.ArrayList;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CalendarDay {
    private int day;
    private List<Dance> dances;

    public void addDance(Dance dance) {
        if (dances == null) {
            dances = new ArrayList<>();
        }
        dances.add(dance);
    }
}