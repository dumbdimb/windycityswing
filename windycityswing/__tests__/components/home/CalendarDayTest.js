import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

import CalendarDay from '..\\..\\..\\src\\scripts\\components\\home\\CalendarDay';

describe('CalendarDay', () => {
    it('renders correctly', () => {
        expect(renderer.create(<CalendarDay day={1}/>).toJSON()).toMatchSnapshot();
    });

    describe('rendering logic', () => {
        describe('when day is not undefined', () => {
            const tree = shallow(<CalendarDay day={1}/>);

            it('has className calendar-day', () => {
                expect(tree.props().className).toEqual('calendar-day');
            });

            it('shows what day it is', () => {
                expect(tree.text()).toEqual('1');
            });
        });

        describe('when day is invalid', () => {
            it('does not render when prop is not specified', () => {
                const tree = shallow(<CalendarDay/>);
                expect(tree.text()).toEqual('');
            });

            it('does not render when day is undefined', () => {
                const tree = shallow(<CalendarDay day={undefined}/>);
                expect(tree.text()).toEqual('');
            });
        });
    });
});