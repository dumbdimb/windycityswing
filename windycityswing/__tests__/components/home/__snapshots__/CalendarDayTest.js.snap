// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarDay renders correctly 1`] = `
<td
  className="calendar-day"
  onClick={[Function]}
>
  <span>
    1
  </span>
  <div
    className="calendar-day__events"
  >
    <div
      className="calendar-day__event event--test-event"
    >
      Test Event
    </div>
  </div>
  <span
    className="calendar-day__call-to-action"
  >
    Details &gt;&gt;
  </span>
</td>
`;
